server/src/graql/internal/reasoner/atom/AtomicFactory.java

```diff
@@ -54,4 +98,237 @@
                 );
     }
 
+    /**
+     * maps a var property to a reasoner atom
+     *
+     * @param varProperty   the var property to be converted to a reasoner atom
+     * @param var           Statement this property belongs to
+     * @param vars          Statements constituting the pattern this property belongs to
+     * @param parent        reasoner query this atom should belong to
+     * @return created atom
+     */
+    private static Atomic mapToAtom(VarProperty varProperty, Statement var, Set<Statement> vars, ReasonerQuery parent) {
```

Comment:
good Lord

Reply:
Haha, I'm sure this will can be refactored into a better shape, @kasper-piskorski. But let's do that once we get to reasoner architecture. Right now we'll have to [carefully] push the mess out of Graql query and pattern classes.

Reply:
It's how it's been in the prototype (0.3?). Definitely not a good place!

Reply:
`AtomicFactory`. That way we break the Open/Closed principle. If we add a new property we need to modify an existing class. Possibly in multiple places.

Reply:
So if a wants to add a new Atom, the developer changes AtomFactory which is used to build Atoms - what's weird in that? 

Don't just look at one specific software engineering principle when designing an architecture. Look at the overall design and cost. As it stands right now: Graql (language) depends on reasoner, Reasoner depends on Transaction, and Transaction depends on Graql (language). This is just one among many circular dependencies that exist in the current code base, @kasper-piskorski.

Start with the higher level architecture that makes sense, then work downwards on deciding the implementation. Everything else is secondary.